monitor/internal/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,135 @@
+package kubernetesmonitor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aporeto-inc/trireme-lib/policy"
+	"go.uber.org/zap"
+	api "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	kubecache "k8s.io/client-go/tools/cache"
+)
+
+// KubernetesPodNameIdentifier is the label used by Docker for the K8S pod name.
+const KubernetesPodNameIdentifier = "@usr:io.kubernetes.pod.name"
+
+// KubernetesPodNamespaceIdentifier is the label used by Docker for the K8S namespace.
+const KubernetesPodNamespaceIdentifier = "@usr:io.kubernetes.pod.namespace"
+
+// KubernetesContainerNameIdentifier is the label used by Docker for the K8S container name.
+const KubernetesContainerNameIdentifier = "@usr:io.kubernetes.container.name"
+
+// KubernetesInfraContainerName is the name of the infra POD.
+const KubernetesInfraContainerName = "POD"
+
+// UpstreamNameIdentifier is the identifier used to identify the nane on the resulting PU
+const UpstreamNameIdentifier = "k8s:name"
+
+// UpstreamNamespaceIdentifier is the identifier used to identify the nanespace on the resulting PU
+const UpstreamNamespaceIdentifier = "k8s:namespace"
+
+func (m *KubernetesMonitor) consolidateKubernetesTags(runtime policy.RuntimeReader) (*policy.PURuntime, error) {
+
+	podName, ok := runtime.Tag(KubernetesPodNameIdentifier)
+	if !ok {
+		return nil, fmt.Errorf("Error getting Kubernetes Pod name")
+	}
+	podNamespace, ok := runtime.Tag(KubernetesPodNamespaceIdentifier)
+	if !ok {
+		return nil, fmt.Errorf("Error getting Kubernetes Pod namespace")
+	}
+
+	pod, err := m.kubernetesClient.Pod(podName, podNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("Couldn't get labels for pod %s : %v", podName, err)
+	}
+
+	// If IP is empty, wait for an UpdatePodEvent with the Actual PodIP. Not ready to be activated now.
+	if pod.Status.PodIP == "" {
+		return nil, nil
+	}
```

Comment:
I was thinking about it. We don't really care if IP is empty. We can use 0.0.0.0/0 .. Its not used anywhere, other than stats.

Reply:
yes I need to change it , it's a remnant from the old Trireme-Kubernetes.

We indeed do not care about the IP, but it is an indicator that the pod is not "ready" yet.

What we need to do now is to cache the Pod//PuRuntime until it is in a ready state to be sent up.